govc/object/find.go

```diff
@@ -0,0 +1,246 @@
+/*
+Copyright (c) 2017 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package object
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"strings"
+
+	"github.com/vmware/govmomi/govc/cli"
+	"github.com/vmware/govmomi/govc/flags"
+	"github.com/vmware/govmomi/property"
+	"github.com/vmware/govmomi/view"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+type find struct {
+	*flags.DatacenterFlag
+
+	ref      bool
+	kind     kinds
+	name     string
+	maxdepth int
+}
+
+type kinds []string
+
+func (e *kinds) String() string {
+	return fmt.Sprint(*e)
+}
+
+func (e *kinds) Set(value string) error {
+	*e = append(*e, value)
+	return nil
+}
+
+func (e *kinds) wanted(kind string) bool {
+	if len(*e) == 0 {
+		return true
+	}
+
+	for _, k := range *e {
+		if kind == k {
+			return true
+		}
+	}
+
+	return false
+}
+
+func init() {
+	cli.Register("object.find", &find{})
```

Comment:
Is there anything other than "objects" to find?  If not, then "object." might be redundant.